lighthouse-core/gather/gather-runner.js

```diff
@@ -197,6 +200,11 @@ class GatherRunner {
                 // Merge pass trace and network data into tracingData.
                 config.trace && Object.assign(tracingData.traces, loadData.traces);
                 config.network && (tracingData.networkRecords = loadData.networkRecords);
+              })
+              .then(_ => {
+                // Update the outer options object of any post-redirect url
+                // Subsequent passes will use the new URL, and it'll be reported outwards
+                options.url = runOptions.url;
```

Comment:
Interesting challenge here. Thee http-redirect gather changes the `options.url` in beforePass and resets in afterPass.

They are separated into separate passes, so we are safe for now, but we need to make sure they don't share a pass. (well, until we fix the http redirect gather)